src/meta-srv/src/procedure/region_migration/downgrade_leader_region.rs

```diff
@@ -181,17 +182,18 @@ impl DowngradeLeaderRegion {
 
         while retry < self.optimistic_retry {
             if let Err(err) = self.downgrade_region(ctx, &instruction).await {
-                if err.is_retryable() {
-                    warn!("Failed to downgrade region, error: {err:?}, retry later.");
-                    sleep(self.retry_initial_interval).await;
-                    retry += 1;
-                } else {
+                if !err.is_retryable() {
                     break;
                 }
+                retry += 1;
+                warn!(
+                    "Failed to downgrade region, error: {err:?}, retry later: {}",
+                    retry < self.optimistic_retry
+                );
+                sleep(self.retry_initial_interval).await;
```

Comment:
When it were not going to retry again(retry == optimistic_retry), so why delay with sleep? 😄

Reply:
Good point 🥵